### PR TITLE
I18n cleanup

### PR DESCRIPTION
**Pull request by jrmajor**
Cleanup like in #2215. Also a bug has been fixed in MainLayoutController.

### CODE REVIEW

File: src/Utils/I18n/I18n.php [hunk 1]
```diff
@@ -1,418 +1,403 @@
...
-namespace {
-
+namespace
+{
     use src\Utils\I18n\I18n;
 
     /**
-     * Return tranlated string
-     *
-     * @param string $str - translation key
```
Comment (kojoty) [inline src/Utils/I18n/I18n.php:16]: Moim zdaniem ten komentarz ma sens - postuluję zostawić go lub dodać coś innego, bardziej opisowego

File: src/Utils/I18n/I18n.php [hunk 2]
```diff
@@ -1,418 +1,403 @@
...
-    use src\Utils\Debug\Debug;
-    use src\Utils\Database\XDb;
     use src\Utils\Cache\OcMemCache;
+    use src\Utils\Database\XDb;
+    use src\Utils\Debug\Debug;
+    use src\Utils\Uri\OcCookie;
+    use src\Utils\Uri\Uri;
 
-    class I18n
+    final class I18n
```
Comment (kojoty) [inline src/Utils/I18n/I18n.php:36]: why "final"?

File: src/Utils/I18n/I18n.php [hunk 3]
```diff
@@ -1,418 +1,403 @@
...
+
+            if (! $instance) {
+                $instance = (new static())->initialize();
             }
 
             return $instance;
         }
 
-        private function initialize()
+        protected function initialize(): self
```
Comment (kojoty) [inline src/Utils/I18n/I18n.php:59]: final class and protected now?

File: src/Utils/I18n/I18n.php [hunk 4]
```diff
@@ -1,418 +1,403 @@
...
-        public static function translatePhrase($translationId, $langCode=null, $skipPostprocess=false, $skipFailoverLang=false)
-        {
-            return self::instance()->translate($translationId, $langCode, $skipPostprocess, $skipFailoverLang);
+        public static function translatePhrase(
+            string $message,
+            string $language = null,
+            bool $skipPostprocess = false,
+            bool $skipFallback = false
+        ): string {
+            return self::instance()->translate(...func_get_args());
```
Comment (kojoty) [inline src/Utils/I18n/I18n.php:129]: hmm... I'm not a supporter of such tricks - I think much more readable is to have the list of args

File: src/Utils/I18n/I18n.php [hunk 5]
```diff
@@ -1,418 +1,403 @@
...
 
         /**
-         * Allow to check if given phrase is present
-         *
-         * @param string $str - phrase to translate
-         * @return boolean
+         * Check if the given message exists.
          */
-        public static function isTranslationAvailable($str)
+        public static function isTranslationAvailable(string $message): bool
```
Comment (kojoty) [inline src/Utils/I18n/I18n.php:135]: minor thing but "message" is rather not a good name for a translation key...

File: src/Utils/I18n/I18n.php [hunk 6]
```diff
@@ -1,418 +1,403 @@
...
+                $translations[substr($key, $prefixLength)] = $this->trArray[$currentLanguage][$key] ?? $text;
             }
 
-            return $t;
+            return $translations;
         }
 
-
         /**
-         * Methods for retrieving and maintaining old-style database translations.
```
Comment (kojoty) [inline src/Utils/I18n/I18n.php:363]: please leave this comment - I think it explains what is going on here

File: src/Utils/I18n/I18n.php (hunk 4 above)
Comment (rapotek) [reply src/Utils/I18n/I18n.php:129]: Accoring to documentation, array spread operator was introduced in 7.4. While there is much work done in adjusting code to 7.4, this operator should not be used until all instances are running on 7.4 or higher versions, IMHO. Are they running in PHP 7.4 by now?

**Comment by kojoty** [comment]
OK, Review is done. Everything looks nice. We can go with it (if it's ready).

File: src/Utils/I18n/I18n.php (hunk 4 above)
Comment (jrmajor) [reply src/Utils/I18n/I18n.php:129]: @kojoty These methods have identical signatures, which are pretty long, so I think it is more readable and makes it clear that this is just an alias for a private method. I'll revert it if you don't like it.

@rapotek It's argument unpacking, introduced in PHP 5.6.

Comment (rapotek) [reply src/Utils/I18n/I18n.php:129]: > @rapotek It's argument unpacking, introduced in PHP 5.6.

Sorry for incorrect interpretation.

**Comment by jrmajor** [comment]
I want to move the `tr()` function to another file with global helpers only, and was waiting for #2213 to be merged to autoload this file, but I can open another PR for that.

File: src/Utils/I18n/I18n.php (hunk 3 above)
Comment (jrmajor) [reply src/Utils/I18n/I18n.php:58]: It seems cleaner to me. Do you want me to revert it?

**Pull request merged**